TransferMethod/Sources/View/Widget/TextWidget.swift

```diff
@@ -111,36 +110,41 @@ class TextWidget: AbstractWidget {
         if let pattern = pattern {
             let currentText = getTextForPatternCharacter(PatternCharacter.lettersAndNumbersPatternCharacter.rawValue,
                                                          inputText)
+            var currentTextIndex = inputText.startIndex
             var finalText = ""
-            var currentIndex = CurrentIndex(textIndex: inputText.startIndex, patternIndex: pattern.startIndex)
             var isEscapedCharacter = false
+            var patternIndex = pattern.startIndex
 
             if let currentText = currentText, !currentText.isEmpty {
                 var patternCharactersToBeWritten = ""
 
                 while true {
-                    let currentPatternCharacter = pattern[currentIndex.patternIndex]
+                    let patternRange = patternIndex ..< pattern.index(after: patternIndex)
+                    let currentPatternCharacter = [Character](pattern[patternRange])
+                    let currentTextRange = currentTextIndex ..< currentText.index(after: currentTextIndex)
+                    let currentTextCharacter = String(currentText[currentTextRange])
 
                     if isEscapedCharacter {
                         isEscapedCharacter = false
-                        finalText += String(currentPatternCharacter)
-                        currentIndex.patternIndex = pattern.index(after: currentIndex.patternIndex)
-                        if currentIndex.patternIndex >= pattern.endIndex
-                            || currentIndex.textIndex >= currentText.endIndex {
+                        finalText += currentPatternCharacter
+                        patternIndex = pattern.index(after: patternIndex)
+                        if patternIndex >= pattern.endIndex || currentTextIndex >= currentText.endIndex {
                             break
                         }
                         continue
                     }
 
-                    applyFormatForPatternCharacter(currentText: currentText,
+                    applyFormatForPatternCharacter(currentPatternCharacter: currentPatternCharacter,
+                                                   currentText: currentText,
+                                                   currentTextCharacter: currentTextCharacter,
+                                                   currentTextIndex: &currentTextIndex,
                                                    finalText: &finalText,
-                                                   currentIndex: &currentIndex,
+                                                   isEscapedCharacter: &isEscapedCharacter,
```

Comment:
this parameters were removed as part of refactor and other code in this class.. something is wrong in this PR